src/components/[guild]/messages/hooks/useUsersXMTPKeys.ts

```diff
@@ -0,0 +1,40 @@
+import useUser from "components/[guild]/hooks/useUser"
+import { useEffect, useState } from "react"
+import { useFetcherWithSign } from "utils/fetcher"
+
+export const useUsersXMTPKeys = () => {
+  const fetcherWithSign = useFetcherWithSign()
+  const [data, setData] = useState(null)
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState(null)
+  const { id, error: userError, isLoading: isUserLoading } = useUser()
+
+  useEffect(() => {
```

Comment:
I think we could just use SWR here, couldn't we? We wouldn't need to handle the isLoading, error, and data states in that case.